File: python/배열의 길이에 따라 다른 연산하기.py
def solution(arr, n):
    answer = []
    for i in range(len(arr)):
        if len(arr) % 2 == 1: 
            if i % 2 == 0:
                answer.append(arr[i]+n)
            else:
                answer.append(arr[i])
        else:
            if i % 2 == 0:
                answer.append(arr[i])
            else:
                answer.append(arr[i]+n)
    return answer

File: java/배열의 길이에 따라 다른 연산하기.java
import java.util.*;

class Solution {
    public int[] solution(int[] arr, int n) {
        List<Integer> answer = new ArrayList<>();
        for(int i = 0; i < arr.length; i++) {
            if(arr.length % 2 == 1){
                if(i % 2 == 0) {
                   answer.add(arr[i]+n); 
                }else {
                    answer.add(arr[i]);
                }
            }else {
                if(i % 2 == 1) {
                    answer.add(arr[i]+n); 
                }else {
                    answer.add(arr[i]);
                }
            }
        }
        int[] result = answer.stream().mapToInt(Integer::intValue).toArray();
        return result;
    }
}